README.md
# tuto1
練習1


history.txt
hatsk/karin/takik/etuk/naok/yurindo/tamak/aya/atkbar/aparelkeik/
masm/masm/yubin/kusamn/dobumn/bakamn/trackseik/kanek/agoware/kakkn/
aypan/yamamin/hirom/ikue/mie/ebis/ebina/ikun/choufcika/mchelmak/
kangsdeb/fukets/yuzdeb/gakujukdeb/gakkideb/vodeb/ginkodeb/ibsknatum/ayasekang/koibar/
tamam/sao/ymkbar/kumibar/okinaw/sugapi/kanapi/miyk/ishimiw/inumiw/
yakuzimasak/feretnana/kaodekatomo/tsudaj/murin/midori/matnae/oimachi/harue/zusism/
icedeb/shoin/kwhisknec/kazehik/sibyamdmsm/kannasm/mushokureik/uenowkg/sosobar/katapi/
gekism/ry/tkhanad/tsukuba/matsdsimi/kofudeb/shitukoi/emideb/1221writer/towermn/
omega/ayudeb/tomodeb/nork/sibmiyu/


plc.js
/* Like a PLC */
$(function(){

$("div.contextual").css("display","none")

$(".version-article").each(function( ndx ){

$(this).css({
"border":"solid 1px steelblue","border-radius":"4px","width":"150px","height":"110px",
"line-height":"98%","padding":"3px","margin":"5px","float":"left",
"overflow":"hidden","white-space":"nowrap","text-overflow":"ellipsis" })

$("header > h3 > a",this).css("font-size","85%")

var $versionoverview = $("div.version-overview",this);
if ( $(this).hasClass("version-closed")) {
	$(this).css("background-color","darkgray")
} else {
	var sdate = $("p:first",$versionoverview).text().match(/(\()(.*)(\))/)
	if ( sdate != null ) {
		var diff = Math.floor((Date.parse(sdate[2]) - (new Date()))/86400000) +1;
		if ( diff < 0 ) { $(this).css({"background-color":"salmon"}) }
			else if ( diff < 3) { $(this).css({"background-color":"yellow","border":"solid 2px red"}) }
			else if ( diff < 7) { $(this).css("background-color","yellow") } 
			else { $(this).css("background-color","lightskyblue") }
		} else {
		$(this).css("background-color","silver")
	}
}

var $firstp = $("p:first", $versionoverview)
if ( $("strong", $firstp).text() == "" ) {
	var str = $firstp.text()
	$firstp.html("(---/--/--)")
	$("<p>" + str + "</p>").insertAfter( $firstp )
} else {
	$("p:first > strong",$versionoverview).css("display","none")
}

$("table.progress",this).css({"width":"100%","align":"center","border":"solid 1px skyblue"})
$("table.progress td.closed",this).css("background-color","steelblue")
$("form",this).css("display","none")

var $progressinfo = $(".progress-info",this)
if ( $progressinfo.text().match(/^.+/) != null ) {
	$progressinfo.css("display","none")
} else {
	var sstr = $progressinfo.html().match( /(<a .*\/a>)(.*\n.*\n.*\()(.*)(\n.*\n\s+)(.*)(\))/ )
	$progressinfo.html( sstr[3] + " / " + sstr[5] + " - " + $progressinfo.prev("p").text() ) 
}

	
});

});

tree.js
function init_tree() {
	// 要素は存在するか
	if ( !$('ul.tree').length) {
		return; // 存在しない場合は終了
	}
	// 展開と折りたたみ
	$('p.tree_controls a.expand_all, p.tree_controls a.collapse_all')
		.click(function() {
			// クラスを調べる
			if ( $(this).hasClass('collapse_all') ) {
				$(this).parent('p').next('ul').find('a.tree_trigger')
				.addClass('trigger_expanded')
				.end().find('ul').addClass('tree_expanded');
				return false;
			} else {
				$(this).parent('p').next('ul').find('a.tree_trigger')
				.removeClass('trigger_expanded')
				.end().find('ul').removeClass('tree_expanded');
			}
			// リンクをたどらない
			this.blur();
			return false;
		});
	
		// ツリーのクリックを待ち受ける
		$('ul.tree a.tree_trigger').live('click', function() {
			// 次のulは非表示か
			if ( $(this).next('ul').is(':hidden')) {
				$(this).removeClass('trigger_expanded').next('ul')
				.removeClass('tree_expanded');
			} else {
				$(this).addClass('trigger_expanded').next('ul')
				.addClass('tree_expanded');
			}
			//リンクをたどらない
			this.blur();
			return false;
		});
		
		// 最後のliにクラスを追加する
		$('ul.tree li:last-child').addClass('last');
		
		// トリガの状態を変更する
		$('ul.tree_expanded').prev('a').addClass('trigger_expanded');
}
	
// 動的な機能を開始
$(document).ready(function() { init_tree(); });


treectrl.js
function init_tree() { 

	if ( !$('div.project-name').length) { 	// Element PROJECT
		return; // 存在しない場合は処理なし
	}
	
	// position:absoluteを解除。上の要素が消えたら上方に詰める動作のため。
	$('div.issue-subject,div.version-name,div.project-name').css({"position":"relative","top":"25px"});
	
	// clicked on ISSUE
		$('div.issue-subject,div.version-name').on('click', function() { 
			
			// 弟要素を調べる
			var $myLeft = parseInt( $(this).css('left') );
			// すぐ下の弟が非表示だったら弟をすべて表示
			var flagHide = $(this).next().is(':hidden') ? true : false;
			
			$(this).nextAll().each( function(i,elem) {
				var $brLeft = parseInt( $(this).css('left') );
				// 奥にいる弟だけ非表示に
				if ( $brLeft > $myLeft ) { 
					if ( flagHide ) {
						$(this).removeClass('div_hidden');
					} else {
						$(this).addClass('div_hidden'); 
					}
					
				} else { exit; }
			});
		});

} 
	 
// 動的な機能を開始 
$(document).ready(function() { init_tree(); }); 


parking_lot_chart2.js
$(function(){

  var $sidebar = $('#sidebar');

  $sidebar.prepend( $('<input id="plc" type="checkbox">PLC</input>').on('change',function() {   
    if ( $(this).prop('checked') ) { plc(); legend(); }
    else { location.reload(); }
  }) )
  
  function legend(){
    $('#content').prepend($('<div id="legend"/>').css({'position':'absolute','top':'125px','left':'200px','width':'80%','height':'25px'})
      .append($('<div style="float:left"><span style="background-color:ivory; border:solid 1px steelblue; width:15px; height:15px; float:left;">&nbsp;</span>未定&nbsp;&nbsp;</div>'))
      .append($('<div style="float:left"><span style="background-color:lightskyblue; width:15px; height:15px; float:left;">&nbsp;</span>余裕&nbsp;&nbsp;</div>'))
      .append($('<div style="float:left"><span style="background-color:yellow; border:solid 1px steelblue;width:15px; height:15px; float:left;">&nbsp;</span>切迫&nbsp;&nbsp;</div>'))
      .append($('<div style="float:left"><span style="background-color:yellow; border:solid 1px red; width:15px; height:15px; float:left;">&nbsp;</span>緊迫&nbsp;&nbsp;</div>'))
      .append($('<div style="float:left"><span style="background-color:salmon; border:solid 1px steelblue;width:15px; height:15px; float:left;">&nbsp;</span>超過&nbsp;&nbsp;</div>'))
      .append($('<div style="float:left"><span style="background-color:tan; border:solid 1px steelblue;width:15px; height:15px; float:left;">&nbsp;</span>終了&nbsp;&nbsp;</div>'))
      .append($('<div><span style="background-color:silver; border:solid 1px steelblue; width:15px; height:15px; float:left;">&nbsp;</span>完了</div>'))
    )
  }

  function plc(){
    $("div.contextual").css("display","none")

    $(".version-article").each(function( ndx ){

      $(this).css({
      "border":"solid 2px steelblue","border-radius":"4px","width":"148px","height":"100px",
      "line-height":"50%","padding":"5px 2px","margin":"5px","float":"left",
      "overflow":"hidden","white-space":"nowrap","text-overflow":"ellipsis" })

      $("header > h3 > a",this).css("font-size","85%")

      var $versionoverview = $("div.version-overview",this);
      var $firstp = $("p:first", $versionoverview)
      if ( $(this).hasClass("version-closed")) {
        $(this).css("background-color","silver") 
      } else if ( null != $firstp.text().match(/^\d{4}\/\d{2}\/\d{2}$/) ) {
        $(this).css("background-color","tan")
        $('p:nth-child(2)',this).append("&nbsp;")
      } else {
        var sdate = $firstp.text().match(/(\()(.*)(\))/)
        if ( sdate != null ) {
          $firstp.text( sdate[2] )
          var diff = Math.floor((Date.parse(sdate[2]) - (new Date()))/86400000) +1;
          if ( diff < 0 ) { $(this).css({"background-color":"salmon"}) }
            else if ( diff < 3) { $(this).css({"background-color":"yellow","border-color":"red"}) }
            else if ( diff < 7) { $(this).css("background-color","yellow") } 
            else { $(this).css("background-color","lightskyblue") }
          } else {
          $(this).css("background-color","ivory")
        }
      }

      if ( $('p:nth-child(2)',this).html() == undefined ) {
        var str = ("" == $firstp.text()) ? "&nbsp;" : $firstp.text();
        $firstp.html("&nbsp;").after( $("<p>" + str + "</p>") )
      }
      $("p:first > strong",$versionoverview).css("display","none")
      $("table.progress",this).css({"width":"100%","align":"center","border":"solid 1px steelblue"})
      $("table.progress td.closed",this).css("background-color","steelblue")
      $("table.progress td.done",this).css("background-color","lightskyblue")
      $("form",this).css("display","none")

      var $progressinfo = $(".progress-info",this)
      if ( $progressinfo.text().match(/^.+/) != null ) {
        $progressinfo.css("display","none")
      } else {
        var sstr = $progressinfo.html().match( /(<a .*\/a>)(.*\n.*\n.*\()(.*)(\n.*\n\s+)(.*)(\))/ )
        $progressinfo.html( $progressinfo.prev("p").text() + " - " + sstr[3] + " / " + sstr[5] ).css({"font-size":"85%","line-height":"120%"})
      }

    }) // end of each
      
  } // function plc
  
});


parking_lot_chart.js
$(function(){

  var toggleFlag = false;

  $('#content').prepend($('<a href="#">[PLC]</a>')
    .css({'position':'absolute','top':'112px','left':'0','width':'10px','height':'10px'})
        .click( function(event){
            if (toggleFlag==false) { plc(); legend(); $(this).text("[roadmap]")} 
            else { location.reload(); }
          toggleFlag = !toggleFlag;
        } )
  )

  function legend(){
    $('#content').prepend($('<div id="legend"/>').css({'position':'absolute','top':'125px','left':'200px','width':'80%','height':'25px'})
      .append($('<div style="float:left"><span style="background-color:ivory; border:solid 1px steelblue; width:15px; height:15px; float:left;">&nbsp;</span>未定&nbsp;&nbsp;</div>'))
      .append($('<div style="float:left"><span style="background-color:lightskyblue; width:15px; height:15px; float:left;">&nbsp;</span>余裕&nbsp;&nbsp;</div>'))
      .append($('<div style="float:left"><span style="background-color:yellow; border:solid 1px steelblue;width:15px; height:15px; float:left;">&nbsp;</span>切迫&nbsp;&nbsp;</div>'))
      .append($('<div style="float:left"><span style="background-color:yellow; border:solid 1px red; width:15px; height:15px; float:left;">&nbsp;</span>緊迫&nbsp;&nbsp;</div>'))
      .append($('<div style="float:left"><span style="background-color:salmon; border:solid 1px steelblue;width:15px; height:15px; float:left;">&nbsp;</span>超過&nbsp;&nbsp;</div>'))
      .append($('<div style="float:left"><span style="background-color:tan; border:solid 1px steelblue;width:15px; height:15px; float:left;">&nbsp;</span>終了&nbsp;&nbsp;</div>'))
      .append($('<div><span style="background-color:silver; border:solid 1px steelblue; width:15px; height:15px; float:left;">&nbsp;</span>完了</div>'))
    )
  }

  function plc(){
    $("div.contextual").css("display","none")

    $(".version-article").each(function( ndx ){

      $(this).css({
      "border":"solid 2px steelblue","border-radius":"4px","width":"148px","height":"100px",
      "line-height":"50%","padding":"5px 2px","margin":"5px","float":"left",
      "overflow":"hidden","white-space":"nowrap","text-overflow":"ellipsis" })

      $("header > h3 > a",this).css("font-size","85%")

      var $versionoverview = $("div.version-overview",this);
      var $firstp = $("p:first", $versionoverview)
      if ( $(this).hasClass("version-closed")) {
        $(this).css("background-color","silver") 
      } else if ( null != $firstp.text().match(/^\d{4}\/\d{2}\/\d{2}$/) ) {
        $(this).css("background-color","tan")
        $('p:nth-child(2)',this).append("&nbsp;")
      } else {
        var sdate = $firstp.text().match(/(\()(.*)(\))/)
        if ( sdate != null ) {
          $firstp.text( sdate[2] )
          var diff = Math.floor((Date.parse(sdate[2]) - (new Date()))/86400000) +1;
          if ( diff < 0 ) { $(this).css({"background-color":"salmon"}) }
            else if ( diff < 3) { $(this).css({"background-color":"yellow","border-color":"red"}) }
            else if ( diff < 7) { $(this).css("background-color","yellow") } 
            else { $(this).css("background-color","lightskyblue") }
          } else {
          $(this).css("background-color","ivory")
        }
      }

      if ( $('p:nth-child(2)',this).html() == undefined ) {
        var str = ("" == $firstp.text()) ? "&nbsp;" : $firstp.text();
        $firstp.html("&nbsp;").after( $("<p>" + str + "</p>") )
      }
      $("p:first > strong",$versionoverview).css("display","none")
      $("table.progress",this).css({"width":"100%","align":"center","border":"solid 1px steelblue"})
      $("table.progress td.closed",this).css("background-color","steelblue")
      $("table.progress td.done",this).css("background-color","lightskyblue")
      $("form",this).css("display","none")

      var $progressinfo = $(".progress-info",this)
      if ( $progressinfo.text().match(/^.+/) != null ) {
        $progressinfo.css("display","none")
      } else {
        var sstr = $progressinfo.html().match( /(<a .*\/a>)(.*\n.*\n.*\()(.*)(\n.*\n\s+)(.*)(\))/ )
        $progressinfo.html( $progressinfo.prev("p").text() + " - " + sstr[3] + " / " + sstr[5] ).css({"font-size":"85%","line-height":"120%"})
      }

    }) // end of each
      
  } // function plc
  
});